ts/image-occlusion/Toolbar.svelte

```diff
@@ -27,8 +31,53 @@ License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
             showAlignTools = false;
         }
     });
+
+    // handle tool changes after initialization
+    $: if (instance && canvas) {
+        disableFunctions();
+        enableSelectable(canvas, true);
+
+        switch (activeTool) {
+            case "magnify":
+                enableSelectable(canvas, false);
+                instance.resume();
+                break;
+            case "draw-rectangle":
+                drawRectangle(canvas);
+                break;
+            case "draw-ellipse":
+                drawEllipse(canvas);
+                break;
+            case "draw-polygon":
+                drawPolygon(canvas, instance);
+                break;
+            default:
+                break;
+        }
+    }
+
+    const disableFunctions = () => {
+        instance.pause();
+        stopDraw(canvas);
+        canvas.selectionColor = "rgba(100, 100, 255, 0.3)";
+    };
 </script>
 
+<div class="tool-bar-container">
+    {#each tools as tool}
+        <IconButton
+            class="tool-icon-button {activeTool == tool.id ? 'active-tool' : ''}"
```

Comment:
Another candidate for class shortcut

Reply:
Getting this error when using class shortcut.
`Classes can only be applied to DOM elements, not componentssvelte(invalid-class)`

Reply:
Whoops, good point.